terraform/modules/relay-app/main.tf

```diff
@@ -0,0 +1,365 @@
+    network = var.vpc_network
+
+    access_config {
+      network_tier = "PREMIUM"
+      # Ephimerical IP address
+    }
+  }
+
+  service_account {
+    email = google_service_account.application.email
+
+    scopes = [
+      # Those are copying gke-default scopes
+      "storage-ro",
+      "logging-write",
+      "monitoring",
+      "service-management",
+      "service-control",
+      "trace",
+      # Required to discover the other instances in the Erlang Cluster
+      "compute-ro",
+    ]
+  }
+
+  metadata = merge({
+    gce-container-declaration = yamlencode({
+      spec = {
+        containers = [{
+          name  = local.application_name != null ? local.application_name : var.image
+          image = "${var.container_registry}/${var.image_repo}/${var.image}:${var.image_tag}"
+          env   = local.environment_variables
+        }]
+
+        volumes = []
+
+        restartPolicy = "Always"
+      }
+    })
+
+    # Enable FluentBit agent for logging, which will be default one from COS 109
+    google-logging-enabled       = "true"
+    google-logging-use-fluentbit = "true"
+
+    # Report health-related metrics to Cloud Monitoring
+    google-monitoring-enabled = "true"
+  })
+
+  depends_on = [
+    google_project_service.compute,
+    google_project_service.pubsub,
+    google_project_service.bigquery,
+    google_project_service.container,
+    google_project_service.stackdriver,
+    google_project_service.logging,
+    google_project_service.monitoring,
+    google_project_service.cloudprofiler,
+    google_project_service.cloudtrace,
+    google_project_service.servicenetworking,
+    google_project_iam_member.artifacts,
+    google_project_iam_member.logs,
+    google_project_iam_member.errors,
+    google_project_iam_member.metrics,
+    google_project_iam_member.service_management,
+    google_project_iam_member.cloudtrace,
+  ]
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+# Create health checks for the application ports
+resource "google_compute_health_check" "port" {
+  project = var.project_id
+
+  name = "${local.application_name}-${var.health_check.name}"
+
+  check_interval_sec  = var.health_check.check_interval_sec != null ? var.health_check.check_interval_sec : 5
+  timeout_sec         = var.health_check.timeout_sec != null ? var.health_check.timeout_sec : 5
+  healthy_threshold   = var.health_check.healthy_threshold != null ? var.health_check.healthy_threshold : 2
+  unhealthy_threshold = var.health_check.unhealthy_threshold != null ? var.health_check.unhealthy_threshold : 2
+
+  log_config {
+    enable = false
+  }
+
+  http_health_check {
+    port = var.health_check.port
+
+    host         = var.health_check.http_health_check.host
+    request_path = var.health_check.http_health_check.request_path
+    response     = var.health_check.http_health_check.response
+  }
+}
+
+# Use template to deploy zonal instance group
+resource "google_compute_instance_group_manager" "application" {
```

Comment:
I am not sure where to specify this but locally, I need the `--init` flag of `docker run` for the app to react to `SIGINT` signal (not sure that is what is used by GCP to shut down applications).

Also, where to we specify that we need to run with docker's host-networking?

Reply:
@thomaseizinger We are running it on GCP by using CoreOS, not sure if they send the `SIGINT` and I don't really have any way to pass it there. I believe all containers run on host network by default so on GCP we rely on firewall rules more than on exposing/not exposing Docker ports. The runtime is also different, it's using `containerd`.`

Reply:
Okay nice!